### PR TITLE
save only non-operation outcome results

### PR DESCRIPTION
**Pull request by czh-orz**
This PR makes it so that sequences don't save OperationOutcome results after performing a search. If the search was successful, and an operationoutcome was returned the bundle, the sequence will continue as if it didn't see an OperationOutcome. 

This change won't affect the situation where a search fails and they return an operation outcome. 

**Submitter:**
- [X] This pull request describes why these changes were made
- [X] Internal ticket for this PR: https://oncprojectracking.healthit.gov/support/browse/FI-1149
- [X] Internal ticket links to this PR
- [X] Internal ticket is properly labeled (Community/Program)
- [X] Internal ticket has a justification for its Community/Program label
- [X] Code diff has been reviewed for extraneous/missing code
- [X] Tests are included and test edge cases
- [X] Tests/code quality metrics have been run locally and pass

**Reviewer 1:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

**Reviewer 2:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

**Comment by arscan** [comment]
We need to explicitly fail if it is something other than the expected resource type prior to validation or must support tabulation.  Because if they do incorrectly still send the wrong resource type, our code ungracefully errors (purple error with cryptic ruby error) instead of stating that it received an unexpected resource type.

Also, I don't think this is right spot to strip this out -- I'd put it closer to the actual test (this seems more like a helper function to page through all resources, I wouldn't expect a function named this to silently remove OperationOutcomes).

File: lib/modules/uscore_v3.1.1/bodyheight_sequence.rb [hunk 1]
```diff
@@ -259,6 +265,10 @@ def patient_ids
           reply = perform_search_with_status(reply, search_params) if reply.code == 400
 
           validate_search_reply(versioned_resource_class('Observation'), reply, search_params)
+          resource_returned = fetch_all_bundled_resources(reply, check_for_data_absent_reasons)
```
Comment (arscan) [inline lib/modules/uscore_v3.1.1/bodyheight_sequence.rb:268]: Is there any way to do this without paging through all the results again?  Our tests are already pretty slow, and this adds more requests.  Can we just move this assert into `validate_search_results`?

File: lib/modules/uscore_v3.1.1/bodyheight_sequence.rb [hunk 2]
```diff
@@ -259,6 +265,10 @@ def patient_ids
...
+          assert(resource_returned.all? { |resource| ['Observation', 'OperationOutcome'].include? resource.resourceType },
+                 'All resources returned must be of the type Observation or OperationOutcome')
+          resource_returned.reject! { |resource| resource.resourceType == 'OperationOutcome' }
```
Comment (arscan) [inline lib/modules/uscore_v3.1.1/bodyheight_sequence.rb:271]: What does `resource_returned` do?  It doesn't appear to be referenced again, so why clear out OperationOutcomes from it?

Comment (arscan) [reply lib/modules/uscore_v3.1.1/bodyheight_sequence.rb:271]: Ah, I think we have it here because it is needed for MedicationRequest.  But I think we could have the `if` statement in the generator be a little more inclusive so that this line doesn't show up in the generated code for no reason for non-MedicaitonRequests.

Comment (arscan) [reply lib/modules/uscore_v3.1.1/bodyheight_sequence.rb:271]: hmm, in looking at the generator code, I can see why it is a little awkward to have a multi-line if block here.

Comment (czh-orz) [reply lib/modules/uscore_v3.1.1/bodyheight_sequence.rb]: Yeah, I changed it to only add this for medication inclusion. I also just moved the all resources are the right resource check into validate_search_reply

**Review by Jammjammjamm — APPROVED** [review]
